test/acl.test.js

```diff
@@ -356,3 +356,42 @@ describe('security ACLs', function() {
     });
   });
 });
+
+describe('access check', function() {
+  var app;
+  before(function() {
+    app = loopback();
+    app.use(loopback.rest());
+    app.enableAuth();
+    app.dataSource('test', {connector: 'memory'});
+  });
+
+  it('should occur before other remote hooks', function(done) {
+    var MyTestModel = app.model('MyTestModel', {base: 'PersistedModel', dataSource: 'test'});
+    var checkAccessCalled = false;
+    var beforeHookCalled = false;
```

Comment:
Please consider using an array of strings instead of boolean properties. Each hook pushes a string name, and then the assertion can verify the order of execution.

```js
var called = [];

MyTestModel.checkAccess = function() {
  called.push('checkAccess');
  // ...
}

// ...

expect(called).to.eql(['checkAccess', 'beforeHook']);
```